File: LC/492.py
import math

def constructRectangle(area):
    """
    Constructs a rectangle with the given area.

    Parameters:
    area (int): The area of the rectangle.

    Returns:
    list: A list containing the length and width of the rectangle.
    """
    width = int(math.sqrt(area))
    while area % width != 0:
        width -= 1
    length = area // width
    return [length, width]


#pass 2

class Solution:
    def constructRectangle(self, area: int) -> List[int]:
        for l in range(int(area**0.5), 0, -1):            
            if area % l == 0: 
                return [area // l, l]

File: LC/492.java
import java.util.ArrayList;
import java.util.List;

public class RectangleConstructor {
    
    // Approach 1: Function
    public static List<Integer> constructRectangle(int area) {
        int width = (int) Math.sqrt(area);
        while (area % width != 0) {
            width--;
        }
        int length = area / width;
        List<Integer> result = new ArrayList<>();
        result.add(length);
        result.add(width);
        return result;
    }

    // Approach 2: Method within a class
    static class Solution {
        public List<Integer> constructRectangle(int area) {
            for (int l = (int) Math.sqrt(area); l > 0; l--) {
                if (area % l == 0) {
                    List<Integer> result = new ArrayList<>();
                    result.add(area / l);
                    result.add(l);
                    return result;
                }
            }
            return null; // This line will never be reached in the current logic
        }
    }

    // Test the code
    public static void main(String[] args) {
        List<Integer> rectangle1 = constructRectangle(4);
        System.out.println("Approach 1: " + rectangle1);

        Solution solution = new Solution();
        List<Integer> rectangle2 = solution.constructRectangle(4);
        System.out.println("Approach 2: " + rectangle2);
    }
}
